gqcp/tests/QCMethod/HF/GHF/QCMethod_GHF_stability_test.cpp

```diff
@@ -49,8 +49,13 @@ BOOST_AUTO_TEST_CASE(H3_stability_test_1) {
     auto ghf_parameters = qc_structure.groundStateParameters();
 
     // We can now check the stability of the ground state parameters.
+    // For this we need a generalized Hamiltonian in the orthonormal MO basis.
+    const auto basis_mo = g_spinor_basis.transformed(ghf_parameters.expansion());
+    const auto h_mo = GQCP::GSQHamiltonian<double>::Molecular(basis_mo, molecule);
```

Comment:
I'd like to propose the following way to generate the generalized molecular Hamiltonian.

```suggestion
    const auto hamiltonian_generalized = sq_hamiltonian.transformed(ghf_parameters.expansion());
```

In this method, there's not really a need for also transforming the spinor basis as well (since you don't use any of its APIs).

Also, I'm not really fond of the undescriptive variable name `h_mo`. I'd rather you change it to something more descriptive, like `hamiltonian_ghf` or something else. The name `h_mo` could also mean the core Hamiltonian in the MO basis, which is why I don't like it.

These comments of course also apply to the RHF and UHF test files.